core/src/main/java/org/apache/cxf/interceptor/FaultOutInterceptor.java

```diff
@@ -81,6 +82,15 @@ public void handleMessage(Message message) {
 
             MessagePartInfo part = fi.getFirstMessagePart();
             DataBinding db = service.getDataBinding();
+            boolean faultDetail = cause instanceof SOAPFaultException
+                    && ((SOAPFaultException)cause).getFault().hasDetail();
+            if (faultDetail) {
+                //the fault detail is already set and we don't need to write
+                //Writes the SoapFaultException to detail will go into an infinite loop because
+                //SoapFaultException's cause field is itself
```

Comment:
I am wondering if we should specifically check for the loop in `SoapFaultException`, following the `getCause()` chain? It seems like otherwise we are cutting off `if (isDOMSupported(db)) {` branch altogether in some cases (not sure how important it is to be fair).

Reply:
This FaultOutInterceptor invocation actually goes from WebFaultOutInterceptor: https://github.com/apache/cxf/blob/master/rt/frontend/jaxws/src/main/java/org/apache/cxf/jaxws/interceptors/WebFaultOutInterceptor.java#L192. There is check for SoapFaultException in WebFaultOutInterceptor